File: yeonwoo/0708_to_0714_week2_robot_regulatory_program/cleaning_robot.js

const Robot = require('./robot')

/**
 * Class with new key and method added to Robot class
 * a class that assign additional keys and values to some robot especially 'cleaning robot'
 * cleaning robot is a robot that moves and cleans some room with size 200 X 200
 *
 * <p>
 *     inherit from robot
 *     additional key = this.cleaningPower
 *          Since all robots do not have the function of 'power', I added only the 'cleaning power' key value.
 * </p>
 *
 * @author Yeonwoo Choi
 * @see see robot
 */
class CleaningRobot extends Robot{
    constructor () {
        super()
        this.cleaningPower = 0
    }

    /**
     * method that assigns value to key
     * @param id
     * @param kind
     * @param name
     * @param x
     * @param y
     * @param price
     * @param distance
     * @param cleaningPower
     */
    setValue (id, kind, name, x, y, price, distance, cleaningPower) {
        this.robotId = id
        this.robotKind = kind
        this.robotName = name
        this.robotLocationX = x
        this.robotLocationY = y
        this.robotPrice = price
        this.robotDistance = distance
        this.cleaningPower = `${this.robotKind}=${cleaningPower}`
    }
}

module.exports = CleaningRobot

File: inseung/week3_class_Robot/CleaningRobot.kt
class CleaningRobot : Robot () {
    override var model = ""
    override var distance = 3
    override var position = Coordinate2D(30, 30)
    override var price = 0
    override var ID = 0

    var cleaningPower = 10

    override fun show () {
        print("CleaningRobot    "); super.show(); print(" cleaningPower = $cleaningPower \n")
    }
    fun cleaningStart () = println("Cleaning Power ${cleaningPower}으로 청소를 시작했습니다")
    fun cleaningStop () = println("Cleaning Power ${cleaningPower}으로 청소를 멈추었습니다")
}